src/java.se/share/data/jdwp/jdwp.spec

```diff
@@ -2147,14 +2147,12 @@ JDWP "Java(tm) Debug Wire Protocol"
                                      "the thread is not alive.")
             (Error INVALID_OBJECT    "Thread or value is not a known ID.")
             (Error THREAD_NOT_SUSPENDED)
-            (Error OPAQUE_FRAME      "Attempted to return early from "
-                                     "a frame corresponding to a native "
-                                     "method. "
-                                     "The thread is a virtual thread and the target "
-                                     "VM is unable force its current frame to return. "
-                                     "Or the implementation is "
-                                     "unable to provide this functionality "
-                                     "on this frame.")
+            (Error OPAQUE_FRAME      "Attempted to return early from a frame "
+                                     "corresponding to a native method, "
+                                     "or the thread is a virtual thread and the target "
+                                     "VM is unable to force its current frame to return, "
+                                     "or the implementation is unable to provide this "
+                                     "functionality on this frame.")
```

Comment:
No objection to combined the sentences but there is an order "or" and a missing comma. I think this is how it should read in list form:

"Attempted to return early from a frame corresponding to a native method, the thread is a virtual thread and the target VM is unable to force its current frame to return, or the implementation is unable to provide this functionality on this frame."